Documents/element_design.md

```diff
@@ -0,0 +1,46 @@
+Designing a UI element for a test automation framework like JDI-Light involves several steps to ensure the element is reusable, robust, and integrates well with the existing framework:
+
+### 1: Identify the Element
+- **Research:** Understand the UI component you want to design. For a material UI design, you might be looking at elements like buttons, cards, dialogs, sliders, etc. Use the JDI-Light, Bootstrap or Material Design guidelines as a reference.
+    - [JDI-Light](https://jdi-docs.github.io/jdi-light/?java#theory)
+    - [Bootstrap](https://getbootstrap.com/docs/5.3/components/) 
+    - [Material 2](https://m2.material.io/components?platform=web)
+    - [Material 3](https://m3.material.io/components?platform=web)
+
+- **Requirement Analysis:** Determine the functionalities and states this element can have. For instance, a button may have different states like hover, disabled, or pressed. For this step it is also good to use Material Design guidelines referencesd above.
+
+### 2: Define the Interface
+- **List Properties:** Decide on the properties your element should expose. For a button, properties could include text, color, size, and state.
+- **Define Methods:** Outline the actions you can perform with the element, e.g., click, getText, isEnabled.
+- **Consider Events:** Think about what events the element should respond to, e.g., onClick, onFocus.
+
+### 3: Consider Inheritance and Abstraction
+- **Base Classes:** Look for any existing elements in JDI-Light that are similar and can act as base classes. This helps in reusing code and keeping the design DRY (Don't Repeat Yourself).
+- **Abstract Elements:** If needed, design an abstract class that holds common properties and methods for a category of elements. For example, all form controls might share certain behaviors.
+
+### 4: Incorporate Element Locators
+- **Locators:** Define how the element will be located in the DOM. Choose the most stable locators to ensure reliable tests.
```

Comment:
I think it is good to mention that we must be carefull with XPath locators and do not forget about . in the beginning in case of usage to be sure that we do not go outside of the root element

Reply:
added